// vocs/_lib/vite/.vocs/dist/assets/addresses-S_l3_8qC.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Contract Addresses",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    li: "li",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "contract-addresses",
        children: ["Contract Addresses", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#contract-addresses",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Complete reference of all FEY Protocol contracts deployed on Base Mainnet."
    }), "\n", jsxs(_components.h2, {
      id: "base-mainnet-chain-id-8453",
      children: ["Base Mainnet (Chain ID: 8453)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#base-mainnet-chain-id-8453",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "core-contracts",
      children: ["Core Contracts", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#core-contracts",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Factory (Fey.sol)"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d",
              children: jsx(_components.code, {
                children: "0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d"
              })
            })
          }), jsx(_components.td, {
            children: "Main factory for token deployments"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "FEY Token"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D",
              children: jsx(_components.code, {
                children: "0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"
              })
            })
          }), jsx(_components.td, {
            children: "Native protocol token (100B supply)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Hook (FeyHookStaticFeeV2)"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x5B409184204b86f708d3aeBb3cad3F02835f68cC",
              children: jsx(_components.code, {
                children: "0x5B409184204b86f708d3aeBb3cad3F02835f68cC"
              })
            })
          }), jsx(_components.td, {
            children: "Uniswap V4 hook for fee management"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h3, {
      id: "supporting-infrastructure",
      children: ["Supporting Infrastructure", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#supporting-infrastructure",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "LP Locker (FeyLpLockerMultiple)"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x975aF6a738f502935AFE64633Ad3EA2A3eb3e7Fa",
              children: jsx(_components.code, {
                children: "0x975aF6a738f502935AFE64633Ad3EA2A3eb3e7Fa"
              })
            })
          }), jsx(_components.td, {
            children: "Manages LP positions and rewards"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Fee Locker (FeyFeeLocker)"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0xf739FC4094F3Df0a1Be08E2925b609F3C3Aa13c6",
              children: jsx(_components.code, {
                children: "0xf739FC4094F3Df0a1Be08E2925b609F3C3Aa13c6"
              })
            })
          }), jsx(_components.td, {
            children: "Stores and distributes protocol fees"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Pool Extension Allowlist"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0xFD549237CdEAbDc14438CAF3f3861e174fDEae46",
              children: jsx(_components.code, {
                children: "0xFD549237CdEAbDc14438CAF3f3861e174fDEae46"
              })
            })
          }), jsx(_components.td, {
            children: "Controls approved pool extensions"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h3, {
      id: "extensions",
      children: ["Extensions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#extensions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Extension"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Purpose"
          })]
        })
      }), jsx(_components.tbody, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "FeyUniv4EthDevBuy"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x173077c319c38bb08D4C4968014357fd518446b4",
              children: jsx(_components.code, {
                children: "0x173077c319c38bb08D4C4968014357fd518446b4"
              })
            })
          }), jsx(_components.td, {
            children: "ETH-based token purchases during deployment"
          })]
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "mev-protection",
      children: ["MEV Protection", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mev-protection",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsx(_components.tbody, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "FeyMevNoop"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x2ebc0fA629b268dFA3d455b67027d507a562EAC0",
              children: jsx(_components.code, {
                children: "0x2ebc0fA629b268dFA3d455b67027d507a562EAC0"
              })
            })
          }), jsx(_components.td, {
            children: "No-op MEV module (template)"
          })]
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "key-addresses",
      children: ["Key Addresses", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#key-addresses",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Role"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Purpose"
          })]
        })
      }), jsx(_components.tbody, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Team Fee Recipient"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x72f5565Ab147105614ca4Eb83ecF15f751Fd8C50",
              children: jsx(_components.code, {
                children: "0x72f5565Ab147105614ca4Eb83ecF15f751Fd8C50"
              })
            })
          }), jsx(_components.td, {
            children: "Receives protocol fees for distribution"
          })]
        })
      })]
    }), "\n", jsxs(_components.h2, {
      id: "pool-information",
      children: ["Pool Information", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#pool-information",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "main-pool-feyweth",
      children: ["Main Pool (FEY/WETH)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#main-pool-feyweth",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Property"
          }), jsx(_components.th, {
            children: "Value"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Pool ID"
            })
          }), jsx(_components.td, {
            children: jsx(_components.code, {
              children: "0xe155c517c53f078f4b443c99436e42c1b80fd2fb1b3508f431c46b8365e4f3f0"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Currency0"
            })
          }), jsxs(_components.td, {
            children: ["WETH (", jsx(_components.code, {
              children: "0x4200000000000000000000000000000000000006"
            }), ")"]
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Currency1"
            })
          }), jsxs(_components.td, {
            children: ["FEY (", jsx(_components.code, {
              children: "0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"
            }), ")"]
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Fee Flag"
            })
          }), jsx(_components.td, {
            children: "Dynamic (0x800000)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Tick Spacing"
            })
          }), jsx(_components.td, {
            children: "200"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Hook"
            })
          }), jsxs(_components.td, {
            children: ["FeyHookStaticFeeV2 (", jsx(_components.code, {
              children: "0x5B409184204b86f708d3aeBb3cad3F02835f68cC"
            }), ")"]
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "external-dependencies",
      children: ["External Dependencies", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#external-dependencies",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "uniswap-v4",
      children: ["Uniswap V4", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#uniswap-v4",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Pool Manager"
            })
          }), jsx(_components.td, {
            children: "TBD"
          }), jsx(_components.td, {
            children: "Uniswap V4 core pool manager"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Position Manager"
            })
          }), jsx(_components.td, {
            children: "TBD"
          }), jsx(_components.td, {
            children: "Uniswap V4 position manager"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h3, {
      id: "permit2",
      children: ["Permit2", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#permit2",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsx(_components.tbody, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Permit2"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x000000000022D473030F116dDEE9F6B43aC78BA3",
              children: jsx(_components.code, {
                children: "0x000000000022D473030F116dDEE9F6B43aC78BA3"
              })
            })
          }), jsx(_components.td, {
            children: "Universal approval interface"
          })]
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "base-infrastructure",
      children: ["Base Infrastructure", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#base-infrastructure",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Contract"
          }), jsx(_components.th, {
            children: "Address"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "WETH"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x4200000000000000000000000000000000000006",
              children: jsx(_components.code, {
                children: "0x4200000000000000000000000000000000000006"
              })
            })
          }), jsx(_components.td, {
            children: "Wrapped ETH on Base"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Superchain Token Bridge"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org/address/0x4200000000000000000000000000000000000010",
              children: jsx(_components.code, {
                children: "0x4200000000000000000000000000000000000010"
              })
            })
          }), jsx(_components.td, {
            children: "Cross-chain token bridge"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "network-details",
      children: ["Network Details", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#network-details",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "base-mainnet",
      children: ["Base Mainnet", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#base-mainnet",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Property"
          }), jsx(_components.th, {
            children: "Value"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Chain ID"
            })
          }), jsx(_components.td, {
            children: "8453"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "RPC URL"
            })
          }), jsx(_components.td, {
            children: jsx(_components.code, {
              children: "https://mainnet.base.org"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Explorer"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://basescan.org",
              children: "BaseScan"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Native Token"
            })
          }), jsx(_components.td, {
            children: "ETH"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Bridge"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://bridge.base.org",
              children: "Base Bridge"
            })
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "contract-verification",
      children: ["Contract Verification", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contract-verification",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "All contracts are verified on BaseScan with source code available:"
    }), "\n", jsxs(_components.h3, {
      id: "factory",
      children: ["Factory", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#factory",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Address"
        }), ": ", jsx(_components.code, {
          children: "0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verification"
        }), ": ", jsx(_components.a, {
          href: "https://basescan.org/address/0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d#code",
          children: "View on BaseScan →"
        })]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "fey-token",
      children: ["FEY Token", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#fey-token",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Address"
        }), ": ", jsx(_components.code, {
          children: "0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verification"
        }), ": ", jsx(_components.a, {
          href: "https://basescan.org/address/0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D#code",
          children: "View on BaseScan →"
        })]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "hook",
      children: ["Hook", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#hook",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Address"
        }), ": ", jsx(_components.code, {
          children: "0x5B409184204b86f708d3aeBb3cad3F02835f68cC"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verification"
        }), ": ", jsx(_components.a, {
          href: "https://basescan.org/address/0x5B409184204b86f708d3aeBb3cad3F02835f68cC#code",
          children: "View on BaseScan →"
        })]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "integration-examples",
      children: ["Integration Examples", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-examples",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "web3-configuration",
      children: ["Web3 Configuration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#web3-configuration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// Base mainnet configuration"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " baseConfig"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  chainId: "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "8453"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  name: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'Base Mainnet'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  rpcUrl: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'https://mainnet.base.org'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  blockExplorer: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'https://basescan.org'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  contracts: {"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    factory: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    feyToken: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    hook: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0x5B409184204b86f708d3aeBb3cad3F02835f68cC'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    feeLocker: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0xf739FC4094F3Df0a1Be08E2925b609F3C3Aa13c6'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    lpLocker: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0x975aF6a738f502935AFE64633Ad3EA2A3eb3e7Fa'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "};"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "contract-instances",
      children: ["Contract Instances", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contract-instances",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "import"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " { ethers } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "from"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " 'ethers'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ";"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// Provider setup"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " provider"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " new"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ethers."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "JsonRpcProvider"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'https://mainnet.base.org'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// Contract instances"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " factory"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " new"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ethers."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Contract"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  '0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  factoryABI,"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  provider"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " feyToken"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " new"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ethers."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Contract"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  '0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  erc20ABI,"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  provider"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " hook"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " new"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ethers."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Contract"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  '0x5B409184204b86f708d3aeBb3cad3F02835f68cC'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  hookABI,"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  provider"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "environment-variables",
      children: ["Environment Variables", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#environment-variables",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Base mainnet configuration"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " CHAIN_ID"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "8453"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " RPC_URL"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"https://mainnet.base.org"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " FACTORY_ADDRESS"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " FEY_TOKEN_ADDRESS"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " HOOK_ADDRESS"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0x5B409184204b86f708d3aeBb3cad3F02835f68cC"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " FEE_LOCKER_ADDRESS"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0xf739FC4094F3Df0a1Be08E2925b609F3C3Aa13c6"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " LP_LOCKER_ADDRESS"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0x975aF6a738f502935AFE64633Ad3EA2A3eb3e7Fa"'
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "cast-commands",
      children: ["Cast Commands", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#cast-commands",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Set environment for easier commands"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " RPC"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"https://mainnet.base.org"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " FACTORY"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " FEY"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Quick status checks"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FACTORY "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"deprecated()(bool)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $RPC"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FEY "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"totalSupply()(uint256)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $RPC"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "security-notes",
      children: ["Security Notes", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#security-notes",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "contract-immutability",
      children: ["Contract Immutability", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contract-immutability",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "Immutable Contracts:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Factory core logic (deployment functions)"
      }), "\n", jsx(_components.li, {
        children: "Token contracts (ERC20 functionality)"
      }), "\n", jsx(_components.li, {
        children: "Hook swap logic (fee collection)"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Upgradeable Components:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "System configuration (admin functions)"
      }), "\n", jsx(_components.li, {
        children: "Module enablement (hooks, lockers, extensions)"
      }), "\n", jsx(_components.li, {
        children: "Fee routing (can be frozen)"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "admin-controls",
      children: ["Admin Controls", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#admin-controls",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "Factory Owner:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Enable/disable system modules"
      }), "\n", jsx(_components.li, {
        children: "Set system-wide configuration"
      }), "\n", jsx(_components.li, {
        children: "Freeze critical parameters"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Token Admins:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Update token metadata"
      }), "\n", jsx(_components.li, {
        children: "Transfer admin role"
      }), "\n", jsx(_components.li, {
        children: "Verify token authenticity"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "access-patterns",
      children: ["Access Patterns", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#access-patterns",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "Public Functions:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Token deployment (", jsx(_components.code, {
          children: "deployToken"
        }), ")"]
      }), "\n", jsxs(_components.li, {
        children: ["Fee claiming (", jsx(_components.code, {
          children: "claimWethFees"
        }), ", ", jsx(_components.code, {
          children: "claimBaseTokenFees"
        }), ")"]
      }), "\n", jsxs(_components.li, {
        children: ["Reward collection (", jsx(_components.code, {
          children: "collectRewards"
        }), ")"]
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Permissioned Functions:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "System configuration (owner/admin only)"
      }), "\n", jsx(_components.li, {
        children: "Module management (owner/admin only)"
      }), "\n", jsx(_components.li, {
        children: "Critical parameter changes (owner only)"
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "related-documentation",
      children: ["Related Documentation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#related-documentation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Contract Details"
        }), ": ", jsx(_components.a, {
          href: "/contracts/factory",
          children: "Core Contracts →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Integration Guide"
        }), ": ", jsx(_components.a, {
          href: "/guides/integration",
          children: "Developer Guides →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "State Queries"
        }), ": ", jsx(_components.a, {
          href: "/guides/cast-commands",
          children: "Cast Commands →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Troubleshooting"
        }), ": ", jsx(_components.a, {
          href: "/support/troubleshooting",
          children: "Support →"
        })]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
