// vocs/_lib/vite/.vocs/dist/assets/faq-DcPrN886.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Frequently Asked Questions",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    br: "br",
    code: "code",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "frequently-asked-questions",
        children: ["Frequently Asked Questions", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#frequently-asked-questions",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Common questions about FEY Protocol, how it works, and how to participate."
    }), "\n", jsxs(_components.h2, {
      id: "general-questions",
      children: ["General Questions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#general-questions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "what-is-fey-protocol",
      children: ["What is FEY Protocol?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-is-fey-protocol",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "FEY Protocol is the first fully permissionless, user-owned launchpad built on Base. Unlike traditional launchpads owned by corporations, FEY operates as decentralized public infrastructure where users gain real ownership through staking $FEY tokens."
    }), "\n", jsx(_components.strong, {
      children: "Key Differences:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "100% of protocol fees redistribute to token stakers"
      }), "\n", jsx(_components.li, {
        children: "No governance required for core operations"
      }), "\n", jsx(_components.li, {
        children: "Users own the infrastructure they use"
      }), "\n", jsx(_components.li, {
        children: "All operations are permissionless and automatic"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "how-is-fey-different-from-pumpfun",
      children: ["How is FEY different from Pump.fun?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-is-fey-different-from-pumpfun",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "While Pump.fun pioneered fair-launch mechanics, it remains a privately-owned corporation:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Feature"
          }), jsx(_components.th, {
            children: "Pump.fun"
          }), jsx(_components.th, {
            children: "FEY Protocol"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Ownership"
            })
          }), jsx(_components.td, {
            children: "Corporate shareholders"
          }), jsx(_components.td, {
            children: "Token stakers (users)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Fee Distribution"
            })
          }), jsx(_components.td, {
            children: "To company treasury"
          }), jsx(_components.td, {
            children: "100% to stakers"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Revenue Sharing"
            })
          }), jsx(_components.td, {
            children: "At company discretion"
          }), jsx(_components.td, {
            children: "Automatic & permanent"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Buybacks"
            })
          }), jsx(_components.td, {
            children: "Optional company decision"
          }), jsx(_components.td, {
            children: "Automatic & public"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Control"
            })
          }), jsx(_components.td, {
            children: "Centralized team"
          }), jsx(_components.td, {
            children: "Permissionless contracts"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h3, {
      id: "what-does-permissionless-mean",
      children: ['What does "permissionless" mean?', jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-does-permissionless-mean",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Permissionless means that core protocol operations happen automatically through smart contracts without requiring approval from any central authority:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Fee distribution is automatic"
      }), "\n", jsx(_components.li, {
        children: "Buybacks are triggered by public functions"
      }), "\n", jsx(_components.li, {
        children: "No governance votes needed for basic operations"
      }), "\n", jsx(_components.li, {
        children: "Anyone can call maintenance functions"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "token-economics",
      children: ["Token Economics", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#token-economics",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "how-does-the-fey-token-work",
      children: ["How does the $FEY token work?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-does-the-fey-token-work",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "$FEY serves multiple roles in the ecosystem:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Base Pairing Token"
      }), ": All deployed tokens pair with $FEY (not ETH)\n", jsx(_components.strong, {
        children: "Staking Token"
      }), ": Stake $FEY to earn protocol fees\n", jsx(_components.strong, {
        children: "Reward Token"
      }), ": Token creators earn fees in $FEY\n", jsx(_components.strong, {
        children: "Governance Token"
      }), ": Future DAO functionality (not yet active)"]
    }), "\n", jsx(_components.strong, {
      children: "Key Properties:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Fixed supply: 100,000,000,000 $FEY (100B tokens)"
      }), "\n", jsx(_components.li, {
        children: "No inflation: No new tokens ever created"
      }), "\n", jsx(_components.li, {
        children: "Burnable: Tokens can be permanently removed"
      }), "\n", jsx(_components.li, {
        children: "Cross-chain: Compatible with Superchain bridges"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "how-do-staking-rewards-work",
      children: ["How do staking rewards work?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-do-staking-rewards-work",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When you stake $FEY, you earn from multiple revenue streams:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Protocol Fees"
      }), ": 20% of all LP fees from every pool\n", jsx(_components.strong, {
        children: "LP Rewards"
      }), ": Fees from protocol-managed liquidity positions", jsx(_components.br, {}), "\n", jsx(_components.strong, {
        children: "Buyback Distribution"
      }), ": $FEY purchased from automatic WETH conversion"]
    }), "\n", jsx(_components.strong, {
      children: "Distribution Mechanism:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Fees accumulate in the Fee Locker contract"
      }), "\n", jsx(_components.li, {
        children: "Stakers can claim their share anytime"
      }), "\n", jsx(_components.li, {
        children: "Rewards are proportional to staked amount"
      }), "\n", jsx(_components.li, {
        children: "Early stakers get higher relative rewards"
      }), "\n"]
    }), "\n", jsxs(_components.aside, {
      "data-callout": "warning",
      children: [jsx(_components.strong, {
        "data-callout-title": true,
        children: "Staking Not Yet Active"
      }), jsx(_components.p, {
        children: "The staking mechanism is still being finalized. For now, hold $FEY tokens in your wallet. Staking will be enabled in a future protocol update."
      })]
    }), "\n", jsxs(_components.h3, {
      id: "what-creates-buy-pressure-for-fey",
      children: ["What creates buy pressure for $FEY?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-creates-buy-pressure-for-fey",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Several mechanisms create consistent demand:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Trading Volume"
      }), ": Every token pairs with $FEY, so all trades involve $FEY\n", jsx(_components.strong, {
        children: "Creator Rewards"
      }), ": Token creators earn fees in $FEY, becoming stakeholders\n", jsx(_components.strong, {
        children: "Automatic Buybacks"
      }), ": WETH fees are converted to $FEY and distributed\n", jsx(_components.strong, {
        children: "Staking Demand"
      }), ": Users need $FEY to earn protocol fees"]
    }), "\n", jsxs(_components.h2, {
      id: "using-the-protocol",
      children: ["Using the Protocol", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#using-the-protocol",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "how-do-i-deploy-a-token-on-fey",
      children: ["How do I deploy a token on FEY?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-do-i-deploy-a-token-on-fey",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Token deployment involves several steps:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Prepare Configuration"
        }), ": Token metadata, liquidity settings, reward splits"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Deploy via Factory"
        }), ": Call ", jsx(_components.code, {
          children: "deployToken()"
        }), " with your configuration"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Include Extensions"
        }), ": Add ETH for dev buy or other extensions"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Automatic Setup"
        }), ": Protocol handles pool creation, liquidity, and MEV protection"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["See our detailed ", jsx(_components.a, {
        href: "/flows/deployment",
        children: "Token Deployment Guide →"
      }), " for complete instructions."]
    }), "\n", jsxs(_components.h3, {
      id: "what-are-extensions",
      children: ["What are extensions?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-are-extensions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Extensions add functionality to token deployments:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "FeyUniv4EthDevBuy"
      }), ": Purchase tokens with ETH during deployment\n", jsx(_components.strong, {
        children: "Future Extensions"
      }), ": Presales, airdrops, vesting, team allocations"]
    }), "\n", jsx(_components.p, {
      children: "Extensions must be approved through the allowlist contract and can be combined in a single deployment."
    }), "\n", jsxs(_components.h3, {
      id: "how-do-creator-rewards-work",
      children: ["How do creator rewards work?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-do-creator-rewards-work",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Token creators earn multiple types of rewards:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Creator Fees"
      }), ": 1% of trading volume in their token\n", jsx(_components.strong, {
        children: "LP Position Rewards"
      }), ": Fees from protocol-managed liquidity\n", jsx(_components.strong, {
        children: "Network Growth"
      }), ": All rewards paid in $FEY, so creators benefit from network success"]
    }), "\n", jsx(_components.strong, {
      children: "Reward Flow:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Trading generates LP fees"
      }), "\n", jsx(_components.li, {
        children: "Creator's share flows to LP Locker"
      }), "\n", jsx(_components.li, {
        children: "LP Locker distributes to configured recipients"
      }), "\n", jsx(_components.li, {
        children: "Recipients can claim from Fee Locker"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "can-i-stake-fey-now",
      children: ["Can I stake $FEY now?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#can-i-stake-fey-now",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Not yet. The staking contract is still being developed. For now:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Hold $FEY in your wallet"
      }), "\n", jsx(_components.li, {
        children: "Monitor for staking contract deployment"
      }), "\n", jsxs(_components.li, {
        children: ["Follow ", jsx(_components.a, {
          href: "https://twitter.com/feyprotocol",
          children: "@feyprotocol"
        }), " for updates"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Once staking is live, you'll be able to:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Stake $FEY through the staking contract"
      }), "\n", jsx(_components.li, {
        children: "Earn fees from all protocol activity"
      }), "\n", jsx(_components.li, {
        children: "Claim rewards through the Fee Locker"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "technical-questions",
      children: ["Technical Questions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#technical-questions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "what-blockchain-is-fey-on",
      children: ["What blockchain is FEY on?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-blockchain-is-fey-on",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["FEY Protocol is deployed on ", jsx(_components.strong, {
        children: "Base Mainnet"
      }), " (Chain ID: 8453)."]
    }), "\n", jsx(_components.strong, {
      children: "Why Base?"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Lower transaction costs than Ethereum"
      }), "\n", jsx(_components.li, {
        children: "Fast block times and finality"
      }), "\n", jsx(_components.li, {
        children: "Growing DeFi ecosystem"
      }), "\n", jsx(_components.li, {
        children: "Superchain compatibility for future expansion"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "what-is-uniswap-v4-and-why-does-fey-use-it",
      children: ["What is Uniswap V4 and why does FEY use it?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-is-uniswap-v4-and-why-does-fey-use-it",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Uniswap V4 is the latest version of the Uniswap protocol with advanced features:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Hooks System"
      }), ": Custom logic for pools (FEY uses this for fee management)\n", jsx(_components.strong, {
        children: "Dynamic Fees"
      }), ": Fees can change based on market conditions\n", jsx(_components.strong, {
        children: "Gas Efficiency"
      }), ": Lower costs for swaps and liquidity operations\n", jsx(_components.strong, {
        children: "Singleton Architecture"
      }), ": All pools share one contract"]
    }), "\n", jsx(_components.p, {
      children: "FEY's hook integrates deeply with V4 to provide:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Automatic fee collection"
      }), "\n", jsx(_components.li, {
        children: "Dynamic fee adjustment"
      }), "\n", jsx(_components.li, {
        children: "MEV protection"
      }), "\n", jsx(_components.li, {
        children: "Extension system integration"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "how-does-mev-protection-work",
      children: ["How does MEV protection work?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-does-mev-protection-work",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "New token pools receive MEV protection during their first 2 minutes:"
    }), "\n", jsx(_components.strong, {
      children: "Protection Features:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "MEV module can increase fees up to 80% to deter MEV bots"
      }), "\n", jsx(_components.li, {
        children: "No liquidity changes allowed during protection window"
      }), "\n", jsx(_components.li, {
        children: "Automatically expires after 2 minutes"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Current Implementation:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: 'Uses a "no-op" module that immediately disables itself'
      }), "\n", jsx(_components.li, {
        children: "Serves as template for future MEV protection strategies"
      }), "\n", jsx(_components.li, {
        children: "Can be upgraded to more sophisticated protection"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "are-the-contracts-audited",
      children: ["Are the contracts audited?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#are-the-contracts-audited",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Contract audit status:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Smart Contract Review"
        }), ": Internal security review completed"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "External Audit"
        }), ": In progress with reputable auditing firm"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Bug Bounty"
        }), ": Community bug bounty program planned"]
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Security Features:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Multiple access controls and permission systems"
      }), "\n", jsx(_components.li, {
        children: "Freeze mechanisms for critical parameters"
      }), "\n", jsx(_components.li, {
        children: "Capped fees and MEV protection limits"
      }), "\n", jsx(_components.li, {
        children: "Reentrancy protection on critical functions"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "how-can-i-verify-contract-authenticity",
      children: ["How can I verify contract authenticity?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-can-i-verify-contract-authenticity",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "All contracts are verified on BaseScan:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check Addresses"
        }), ": Use our ", jsx(_components.a, {
          href: "/reference/addresses",
          children: "official address list →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verify Source Code"
        }), ": All contracts have verified source on BaseScan"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Token Verification"
        }), ": Official $FEY token has verification status"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Social Verification"
        }), ": Follow ", jsx(_components.a, {
          href: "https://twitter.com/feyprotocol",
          children: "@feyprotocol"
        }), " for official announcements"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "troubleshooting",
      children: ["Troubleshooting", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#troubleshooting",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "why-cant-i-deploy-a-token",
      children: ["Why can't I deploy a token?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#why-cant-i-deploy-a-token",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Common issues and solutions:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Factory Deprecated"
      }), ": Factory may be temporarily disabled"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.em, {
          children: "Solution"
        }), ": Wait for reactivation or check if you need bootstrap access"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Base Token Not Set"
      }), ": Protocol not fully initialized"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.em, {
          children: "Solution"
        }), ": Wait for protocol bootstrap completion"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Extension Not Enabled"
      }), ": Extension not approved in allowlist"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.em, {
          children: "Solution"
        }), ": Contact team to add extension to allowlist"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Configuration Errors"
      }), ": Invalid deployment configuration"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.em, {
          children: "Solution"
        }), ": Check that all arrays have matching lengths and reward BPS sum to 10,000"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "where-are-my-creator-rewards",
      children: ["Where are my creator rewards?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#where-are-my-creator-rewards",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Rewards flow through several contracts:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check LP Locker"
        }), ": Ensure ", jsx(_components.code, {
          children: "collectRewards()"
        }), " is being called for your token"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check Fee Locker"
        }), ": Query ", jsx(_components.code, {
          children: "availableFees()"
        }), " for your address"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Claim Rewards"
        }), ": Call ", jsx(_components.code, {
          children: "claim()"
        }), " function to withdraw available fees"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["Use our ", jsx(_components.a, {
        href: "/guides/cast-commands",
        children: "Cast commands →"
      }), " to check reward status."]
    }), "\n", jsxs(_components.h3, {
      id: "why-isnt-my-extension-working",
      children: ["Why isn't my extension working?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#why-isnt-my-extension-working",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Extension troubleshooting:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check Allowlist"
        }), ": Verify extension is enabled in allowlist contract"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check Pool Configuration"
        }), ": Ensure extension is properly registered"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check Extension Data"
        }), ": Verify extension data is correctly encoded"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check ETH Amount"
        }), ": Ensure sufficient ETH sent for extension"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "how-do-i-get-support",
      children: ["How do I get support?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-do-i-get-support",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "Community Support:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Discord: ", jsx(_components.a, {
          href: "https://discord.gg/feyprotocol",
          children: "Join our community"
        })]
      }), "\n", jsxs(_components.li, {
        children: ["Twitter: ", jsx(_components.a, {
          href: "https://twitter.com/feyprotocol",
          children: "@feyprotocol"
        })]
      }), "\n", jsxs(_components.li, {
        children: ["GitHub: ", jsx(_components.a, {
          href: "https://github.com/feyprotocol",
          children: "Issues and discussions"
        })]
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Documentation:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/support/troubleshooting",
          children: "Troubleshooting Guide →"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/guides/integration",
          children: "Developer Guides →"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/contracts/factory",
          children: "Contract Documentation →"
        })
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "For Emergencies:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Security issues: ", jsx(_components.a, {
          href: "mailto:security@feyprotocol.xyz",
          children: "security@feyprotocol.xyz"
        })]
      }), "\n", jsx(_components.li, {
        children: "Critical bugs: Please report responsibly through proper channels"
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "still-have-questions",
      children: ["Still have questions?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#still-have-questions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If you didn't find what you're looking for:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check our guides"
        }), ": ", jsx(_components.a, {
          href: "/getting-started",
          children: "Getting Started →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Read the docs"
        }), ": ", jsx(_components.a, {
          href: "/architecture/overview",
          children: "Architecture Overview →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Join the community"
        }), ": ", jsx(_components.a, {
          href: "https://discord.gg/feyprotocol",
          children: "Discord"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Follow updates"
        }), ": ", jsx(_components.a, {
          href: "https://twitter.com/feyprotocol",
          children: "@feyprotocol"
        })]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
