// vocs/_lib/vite/.vocs/dist/assets/index-CMH5M5-a.js
import { jsx, jsxs } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
import "react";
import { L as Link, b as button$1, a as button_accent, u as useConfig, c as Logo$1, R as Root$1, d as List, T as Trigger, C as Content } from "../index.server.js";
import clsx from "clsx";
import "nuqs/adapters/react";
import "react-router";
import "node:fs";
import "node:path";
import "toml";
import "vite";
import "@vanilla-extract/dynamic";
import "react-intersection-observer";
import "@mdx-js/mdx";
import "@radix-ui/react-icons";
import "@radix-ui/react-dialog";
import "nuqs";
import "minisearch";
import "@radix-ui/react-label";
import "mark.js";
import "@radix-ui/react-navigation-menu";
import "@radix-ui/react-accordion";
import "radix-ui";
import "@radix-ui/react-popover";
import "@mdx-js/react";
import "@radix-ui/react-tabs";
import "@floating-ui/react";
import "picomatch";
function Button$1({ children, className, href, variant }) {
  return jsx(Link, { className: clsx(className, button$1, variant === "accent" && button_accent), href, variant: "styleless", children });
}
var root = "vocs_HomePage";
var logo = "vocs_HomePage_logo";
var title = "vocs_HomePage_title";
var tagline = "vocs_HomePage_tagline";
var description = "vocs_HomePage_description";
var buttons = "vocs_HomePage_buttons";
var button = "vocs_HomePage_button";
var tabs = "vocs_HomePage_tabs";
var tabsList = "vocs_HomePage_tabsList";
var tabsContent = "vocs_HomePage_tabsContent";
var packageManager = "vocs_HomePage_packageManager";
function Root({ children, className }) {
  return jsx("div", { className: clsx(className, root), children });
}
function Logo({ className }) {
  const { logoUrl, title: title$1 } = useConfig();
  return logoUrl ? jsx("div", { className: clsx(className, logo), children: jsx(Logo$1, {}) }) : jsx("h1", { className: clsx(className, title), children: title$1 });
}
function Tagline({ children, className }) {
  return jsx("div", { className: clsx(className, tagline), children });
}
function Description({ children, className }) {
  return jsx("div", { className: clsx(className, description), children });
}
function Buttons({ children, className }) {
  return jsx("div", { className: clsx(className, buttons), children });
}
function Button(props) {
  return jsx(Button$1, { ...props, className: clsx(button, props.className) });
}
function InstallPackage({ name, type = "install" }) {
  return jsxs(Root$1, { className: tabs, defaultValue: "npm", children: [jsxs(List, { className: tabsList, children: [jsx(Trigger, { value: "npm", children: "npm" }), jsx(Trigger, { value: "pnpm", children: "pnpm" }), jsx(Trigger, { value: "yarn", children: "yarn" })] }), jsxs(Content, { className: tabsContent, value: "npm", children: [jsx("span", { className: packageManager, children: "npm" }), " ", type === "init" ? "init" : "install", " ", name] }), jsxs(Content, { className: tabsContent, value: "pnpm", children: [jsx("span", { className: packageManager, children: "pnpm" }), " ", type === "init" ? "create" : "add", " ", name] }), jsxs(Content, { className: tabsContent, value: "yarn", children: [jsx("span", { className: packageManager, children: "yarn" }), " ", type === "init" ? "create" : "add", " ", name] })] });
}
const HomePage = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Button,
  Buttons,
  Description,
  InstallPackage,
  Logo,
  Root,
  Tagline
}, Symbol.toStringTag, { value: "Module" }));
const frontmatter = {
  "layout": "landing"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h3: "h3",
    li: "li",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  if (!HomePage) _missingMdxReference("HomePage", false);
  if (!Button) _missingMdxReference("HomePage.Button", true);
  if (!Buttons) _missingMdxReference("HomePage.Buttons", true);
  if (!Description) _missingMdxReference("HomePage.Description", true);
  if (!Logo) _missingMdxReference("HomePage.Logo", true);
  if (!Root) _missingMdxReference("HomePage.Root", true);
  if (!Tagline) _missingMdxReference("HomePage.Tagline", true);
  return jsxs(Root, {
    children: [jsx(Logo, {}), jsx(Tagline, {
      children: "The First Fully Permissionless, User-Owned Launchpad"
    }), jsx(Description, {
      children: jsx(_components.p, {
        children: "FEY Protocol is built on Base and redistributes 100% of protocol fees to token stakers.\nUnlike traditional launchpads owned by corporations, FEY operates as decentralized public\ninfrastructure where users gain real ownership through staking."
      })
    }), jsxs("div", {
      style: {
        margin: "2rem 0"
      },
      children: [jsxs(_components.h3, {
        id: "core-features",
        children: ["Core Features", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#core-features",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsxs(_components.ul, {
        children: ["\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "100% Fee Redistribution"
          }), ": All protocol fees automatically flow back to stakers"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Permissionless Operations"
          }), ": No governance required for core protocol functions"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Creator Rewards"
          }), ": Token deployers earn fees in $FEY for successful launches"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Automatic Buybacks"
          }), ": WETH fees trigger $FEY buybacks every 60 seconds"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "TOKEN/FEY Pairing"
          }), ": All tokens pair with $FEY, creating constant buy pressure"]
        }), "\n"]
      })]
    }), jsxs(Buttons, {
      children: [jsx(Button, {
        href: "/introduction",
        variant: "accent",
        children: "Learn About FEY"
      }), jsx(Button, {
        href: "/getting-started",
        children: "Quick Start"
      }), jsx(Button, {
        href: "/architecture/overview",
        children: "Architecture"
      }), jsx(Button, {
        href: "https://github.com/feyprotocol",
        children: "GitHub"
      })]
    }), jsxs("div", {
      style: {
        margin: "3rem 0 2rem 0",
        padding: "2rem",
        backgroundColor: "#1a1a1a",
        borderRadius: "8px"
      },
      children: [jsxs(_components.h3, {
        id: "network-information",
        children: ["Network Information", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#network-information",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsx(_components.strong, {
        children: "Base Mainnet Deployment"
      }), jsxs(_components.ul, {
        children: ["\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Factory"
          }), ": ", jsx(_components.code, {
            children: "0x8EEF0dC80ADf57908bB1be0236c2a72a7e379C2d"
          })]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "FEY Token"
          }), ": ", jsx(_components.code, {
            children: "0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"
          })]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Total Supply"
          }), ": 100,000,000,000 FEY (100B tokens)"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Main Pool"
          }), ": FEY/WETH on Uniswap V4"]
        }), "\n"]
      }), jsx(_components.p, {
        children: jsx(_components.a, {
          href: "/reference/addresses",
          children: "View all contract addresses →"
        })
      })]
    }), jsxs("div", {
      style: {
        margin: "2rem 0",
        display: "grid",
        gridTemplateColumns: "repeat(auto-fit, minmax(250px, 1fr))",
        gap: "1rem"
      },
      children: [jsxs("div", {
        style: {
          padding: "1.5rem",
          border: "1px solid #333",
          borderRadius: "8px"
        },
        children: [jsx(_components.strong, {
          children: "🏭 For Token Creators"
        }), jsx(_components.p, {
          children: "Launch your token on FEY and earn creator rewards in $FEY from trading fees. Fair launch mechanics with MEV protection."
        }), jsx(_components.p, {
          children: jsx(_components.a, {
            href: "/flows/deployment",
            children: "Deploy a Token →"
          })
        })]
      }), jsxs("div", {
        style: {
          padding: "1.5rem",
          border: "1px solid #333",
          borderRadius: "8px"
        },
        children: [jsx(_components.strong, {
          children: "💰 For Stakers"
        }), jsx(_components.p, {
          children: "Stake $FEY to receive 100% of protocol fees. Automatic buybacks and reward distribution through permissionless contracts."
        }), jsx(_components.p, {
          children: jsx(_components.a, {
            href: "/architecture/tokenomics",
            children: "Learn About Staking →"
          })
        })]
      }), jsxs("div", {
        style: {
          padding: "1.5rem",
          border: "1px solid #333",
          borderRadius: "8px"
        },
        children: [jsx(_components.strong, {
          children: "🔧 For Developers"
        }), jsx(_components.p, {
          children: "Build on FEY with extensions, integrate with our APIs, or contribute to the protocol. Full technical documentation available."
        }), jsx(_components.p, {
          children: jsx(_components.a, {
            href: "/guides/integration",
            children: "Developer Guides →"
          })
        })]
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
function _missingMdxReference(id, component) {
  throw new Error("Expected " + (component ? "component" : "object") + " `" + id + "` to be defined: you likely forgot to import, pass, or provide it.");
}
export {
  MDXContent as default,
  frontmatter
};
