// vocs/_lib/vite/.vocs/dist/assets/introduction-BqaFoGnM.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "What is FEY Protocol?",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    li: "li",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "what-is-fey-protocol",
        children: ["What is FEY Protocol?", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#what-is-fey-protocol",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "FEY Protocol is the first fully permissionless, user-owned launchpad built on Base. Unlike traditional launchpads owned by corporations, FEY operates as decentralized public infrastructure where users gain real ownership through staking."
    }), "\n", jsxs(_components.h2, {
      id: "the-vision",
      children: ["The Vision", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#the-vision",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.em, {
        children: '"Everything will be tokenized"'
      }), " - this has been a thesis in crypto for years. Pumpfun proved this out as the first successful fair-launch platform, becoming one of the most profitable onchain experiments ever, reaching $750M revenue faster than any other protocol."]
    }), "\n", jsxs(_components.p, {
      children: ["However, all existing launchpads share a fundamental problem: ", jsx(_components.strong, {
        children: "they are privately-owned corporations"
      }), ". While built on crypto rails for crypto participants, their structure and ownership remain from the old world."]
    }), "\n", jsxs(_components.p, {
      children: ["FEY changes this by creating the first ", jsx(_components.strong, {
        children: "user-owned launchpad network"
      }), " where:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "100% of protocol fees redistribute to token holders"
      }), "\n", jsx(_components.li, {
        children: "Operations are completely permissionless"
      }), "\n", jsx(_components.li, {
        children: "Users get real ownership, not just utility tokens"
      }), "\n", jsx(_components.li, {
        children: "The network is self-sustaining without external capital"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "how-fey-works",
      children: ["How FEY Works", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-fey-works",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "core-mechanics",
      children: ["Core Mechanics", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#core-mechanics",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "🔄 Automatic Fee Distribution"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "20% of all LP fees flow to the protocol"
      }), "\n", jsx(_components.li, {
        children: "Fees are collected in the paired token (FEY for most pools, WETH for the base pool)"
      }), "\n", jsx(_components.li, {
        children: "100% redistribution to stakers through permissionless contracts"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "💎 TOKEN/FEY Pairing"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "All deployed tokens pair with $FEY (similar to Zora's model)"
      }), "\n", jsx(_components.li, {
        children: "Creates constant buy pressure on $FEY from every trade"
      }), "\n", jsx(_components.li, {
        children: "Token creators earn rewards in $FEY, aligning them with network growth"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "⚡ Permissionless Buybacks"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "WETH fees trigger automatic $FEY buybacks"
      }), "\n", jsx(_components.li, {
        children: "Anyone can call the buyback function (typically every 60 seconds)"
      }), "\n", jsx(_components.li, {
        children: "Purchased $FEY is distributed to active stakers"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "🛡️ MEV Protection"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "New pools get 2-minute MEV protection window"
      }), "\n", jsx(_components.li, {
        children: "Prevents sniping during fair launches"
      }), "\n", jsx(_components.li, {
        children: "Automatically disables to allow normal trading"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "network-participants",
      children: ["Network Participants", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#network-participants",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.strong, {
      children: "Token Creators"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Deploy tokens on FEY and earn creator rewards"
      }), "\n", jsx(_components.li, {
        children: "Receive 1% of trading fees in $FEY"
      }), "\n", jsx(_components.li, {
        children: "Gain exposure to network growth through $FEY holdings"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Stakers"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Stake $FEY to receive protocol fee distributions"
      }), "\n", jsx(_components.li, {
        children: "Earn from all trading activity across the network"
      }), "\n", jsx(_components.li, {
        children: "True ownership stake in launchpad infrastructure"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Traders"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Access to fair-launch tokens with MEV protection"
      }), "\n", jsx(_components.li, {
        children: "Trade on Uniswap V4 with dynamic fee optimization"
      }), "\n", jsx(_components.li, {
        children: "Benefit from deep liquidity provided by the protocol"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "key-differences-from-other-launchpads",
      children: ["Key Differences from Other Launchpads", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#key-differences-from-other-launchpads",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Feature"
          }), jsx(_components.th, {
            children: "Traditional Launchpads"
          }), jsx(_components.th, {
            children: "FEY Protocol"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Ownership"
            })
          }), jsx(_components.td, {
            children: "Corporate shareholders"
          }), jsx(_components.td, {
            children: "Token stakers (users)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Fee Distribution"
            })
          }), jsx(_components.td, {
            children: "To company treasury"
          }), jsx(_components.td, {
            children: "100% to stakers"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Revenue Sharing"
            })
          }), jsx(_components.td, {
            children: "At company discretion"
          }), jsx(_components.td, {
            children: "Automatic & permanent"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Governance"
            })
          }), jsx(_components.td, {
            children: "Centralized decisions"
          }), jsx(_components.td, {
            children: "Permissionless operations"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Creator Rewards"
            })
          }), jsx(_components.td, {
            children: "Fiat or company tokens"
          }), jsx(_components.td, {
            children: "Network token ($FEY)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Buybacks"
            })
          }), jsx(_components.td, {
            children: "Optional company decision"
          }), jsx(_components.td, {
            children: "Automatic & public"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "why-this-matters",
      children: ["Why This Matters", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#why-this-matters",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "FEY represents a return to crypto's founding principles:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Self-sovereignty"
        }), ": Users own the infrastructure they use"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Permissionless"
        }), ": No central authority controls operations"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Aligned incentives"
        }), ": All participants benefit from network growth"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Public goods"
        }), ": Infrastructure owned by its users, not corporations"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "When you participate in FEY, you're not just using a launchpad - you're becoming a co-owner of the network itself."
    }), "\n", jsxs(_components.h2, {
      id: "getting-started",
      children: ["Getting Started", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#getting-started",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Ready to explore FEY Protocol?"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Learn the concepts"
        }), ": ", jsx(_components.a, {
          href: "/concepts",
          children: "Key Concepts →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Start using"
        }), ": ", jsx(_components.a, {
          href: "/getting-started",
          children: "Getting Started Guide →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Deep dive"
        }), ": ", jsx(_components.a, {
          href: "/architecture/overview",
          children: "Protocol Architecture →"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "For developers"
        }), ": ", jsx(_components.a, {
          href: "/guides/integration",
          children: "Integration Guide →"
        })]
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "FEY Protocol: Where users own the rails they trade on."
      })
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
