// vocs/_lib/vite/.vocs/dist/assets/queries-B83MMAEV.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Queries",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "queries",
        children: ["Queries", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#queries",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "This page provides guidance on querying the Fey protocol."
    }), "\n", jsxs(_components.h2, {
      id: "coming-soon",
      children: ["Coming Soon", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#coming-soon",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Detailed query documentation will be available here."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
