// vocs/_lib/vite/.vocs/dist/assets/troubleshooting-DHb48nZ5.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Troubleshooting",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    input: "input",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "troubleshooting",
        children: ["Troubleshooting", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#troubleshooting",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Common issues and solutions when working with FEY Protocol."
    }), "\n", jsxs(_components.h2, {
      id: "deployment-issues",
      children: ["Deployment Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#deployment-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "factory-deprecated-error",
      children: ["Factory Deprecated Error", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#factory-deprecated-error",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ': Getting "Deprecated" error when trying to deploy tokens.']
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Cause"
      }), ": The factory is temporarily disabled for new deployments."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check factory status"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FACTORY "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"deprecated()(bool)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " --rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Wait for reactivation"
          }), ": Factory may be temporarily disabled for upgrades"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Bootstrap access"
          }), ": If you have bootstrap permissions, you can still deploy"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check announcements"
          }), ": Follow ", jsx(_components.a, {
            href: "https://twitter.com/feyprotocol",
            children: "@feyprotocol"
          }), " for status updates"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "base-token-not-set",
      children: ["Base Token Not Set", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#base-token-not-set",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ': Getting "BaseTokenNotSet" error.']
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Cause"
      }), ": Protocol hasn't completed initial bootstrap process."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check if base token is set"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FACTORY "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"baseToken()(address)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Should return FEY token address: 0xD09cf0982A32DD6856e12d6BF2F08A822eA5D91D"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# If it returns 0x000...000, protocol isn't ready"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solution"
      }), ": Wait for protocol bootstrap completion or contact team."]
    }), "\n", jsxs(_components.h3, {
      id: "invalid-reward-configuration",
      children: ["Invalid Reward Configuration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#invalid-reward-configuration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Deployment reverts with reward-related errors."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Common Causes"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Reward BPS don't sum to 10000"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ❌ Wrong - sums to 9000"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "4000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ✅ Correct - sums to 10000  "
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "6000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "4000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Array length mismatch"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ❌ Wrong - different lengths"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "],"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardAdmins"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x1234"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "],  "
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Only 1 element"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardRecipients"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x1234"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x5678"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]  "
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// 2 elements"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ✅ Correct - matching lengths"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "],"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardAdmins"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x1234"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x5678"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "],"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "rewardRecipients"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x1234"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0x5678"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Position BPS don't sum to 10000"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ❌ Wrong"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "positionBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "3000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]  "
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Only 8000"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ✅ Correct"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "positionBps"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": ["
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "7000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "3000"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]  "
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Equals 10000"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "extension-configuration-errors",
      children: ["Extension Configuration Errors", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#extension-configuration-errors",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Extension-related deployment failures."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnostics"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check if extension is enabled"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 0xFD549237CdEAbDc14438CAF3f3861e174fDEae46"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  "enabledExtensions(address)(bool)"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " [EXTENSION_ADDRESS] "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: "\\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "  --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check dev buy extension specifically"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 0xFD549237CdEAbDc14438CAF3f3861e174fDEae46"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  "enabledExtensions(address)(bool)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 0x173077c319c38bb08D4C4968014357fd518446b4"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Common Fixes"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Extension not enabled"
          }), ": Contact team to add extension to allowlist"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Invalid extension data"
          }), ": Check data encoding:"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// For dev buy extension"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "const"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " extensionData"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " ="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " ethers.AbiCoder."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "defaultAbiCoder"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "()."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "encode"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  ["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "'address'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "'tuple(address,address,uint24,int24,address)'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "'uint128'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "],"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  [recipient, poolKey, minAmountOut]"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ");"
                })
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "ETH amount mismatch"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Ensure ETH sent matches extension configs"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "const"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " totalEth"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " ="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " extensionConfigs."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "reduce"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  ("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "sum"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ext"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "=>"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " sum "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "+"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: " BigInt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(ext.msgValue),"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "  0"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "n"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ");"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Send exactly this amount with deployment"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "await"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " factory."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "deployToken"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(config, { value: totalEth });"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mev-module-issues",
      children: ["MEV Module Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mev-module-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": MEV module related errors during deployment."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnostics"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check if MEV module is enabled"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FACTORY "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"enabledMevModules(address)(bool)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  0x2ebc0fA629b268dFA3d455b67027d507a562EAC0"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["Use the standard no-op MEV module: ", jsx(_components.code, {
          children: "0x2ebc0fA629b268dFA3d455b67027d507a562EAC0"
        })]
      }), "\n", jsx(_components.li, {
        children: "Ensure MEV module data is properly encoded"
      }), "\n", jsx(_components.li, {
        children: "Contact team if you need custom MEV protection"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "fee-and-reward-issues",
      children: ["Fee and Reward Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#fee-and-reward-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "no-rewards-appearing",
      children: ["No Rewards Appearing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#no-rewards-appearing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Not receiving expected creator or staking rewards."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis Steps"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check token reward configuration"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $LP_LOCKER "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"tokenRewards(address)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " [TOKEN_ADDRESS] --rpc-url https://mainnet.base.org"
                })]
              })
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check if rewards collection is being triggered"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "# Anyone can trigger this"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " send"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $LP_LOCKER "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"collectRewards(address)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " [TOKEN_ADDRESS] "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "\\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "  --private-key"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $PRIVATE_KEY "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "--rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check your available fees"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FEE_LOCKER "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "\\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '  "availableFees(address,address)(uint256)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: " \\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  [YOUR_ADDRESS] [TOKEN_ADDRESS] "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "\\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "  --rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Common Causes"
      }), ":"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Rewards collection not being called regularly"
      }), "\n", jsx(_components.li, {
        children: "Low trading volume (no fees generated)"
      }), "\n", jsx(_components.li, {
        children: "Incorrect recipient address in deployment config"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "fee-claims-failing",
      children: ["Fee Claims Failing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#fee-claims-failing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Cannot claim fees from Fee Locker."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check if you have fees to claim"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FEE_LOCKER "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: "\\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  "availableFees(address,address)(uint256)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: " \\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  [YOUR_ADDRESS] [TOKEN_ADDRESS] "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#F47067"
              },
              children: "\\"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "  --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check if address is authorized (for vault claims)"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $FEE_LOCKER "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"vault()(address)"'
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --rpc-url"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://mainnet.base.org"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "No fees available"
        }), ": Wait for more trading activity or rewards distribution"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Vault restrictions"
        }), ": If claiming for vault address, must call from vault itself"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Wrong token"
        }), ": Ensure you're checking the correct token address"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "protocol-fees-not-accumulating",
      children: ["Protocol Fees Not Accumulating", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#protocol-fees-not-accumulating",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Expected protocol fees not reaching team recipient."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis Flow"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check hook fee extraction"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "# Verify hook is properly configured"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $HOOK "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"protocolFee()(uint24)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " --rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check factory balances"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "# Factory should accumulate fees temporarily"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FEY_TOKEN "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"balanceOf(address)(uint256)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FACTORY "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "--rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " 0x4200000000000000000000000000000000000006"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: ' "balanceOf(address)(uint256)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FACTORY "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "--rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Trigger fee claims"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "# Anyone can call these"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " send"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FACTORY "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"claimWethFees()"'
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " --private-key"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $PRIVATE_KEY "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "--rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " send"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $FACTORY "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"claimBaseTokenFees()"'
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " --private-key"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $PRIVATE_KEY "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "--rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "pool-and-trading-issues",
      children: ["Pool and Trading Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#pool-and-trading-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "pool-extensions-not-working",
      children: ["Pool Extensions Not Working", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#pool-extensions-not-working",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Pool extensions not executing during swaps."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check extension registration"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $HOOK "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"poolExtension(bytes32)(address)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " [POOL_ID] --rpc-url https://mainnet.base.org"
                })]
              })
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check if extension is set up"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " $HOOK "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"poolExtensionSetup(bytes32)(bool)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " [POOL_ID] --rpc-url https://mainnet.base.org"
                })]
              })
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Verify allowlist"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "cast"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " call"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " 0xFD549237CdEAbDc14438CAF3f3861e174fDEae46"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: " \\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '  "enabledExtensions(address)(bool)"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " [EXTENSION_ADDRESS] "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "\\"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "  --rpc-url"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " https://mainnet.base.org"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Extensions only work for factory-deployed pools"
      }), "\n", jsx(_components.li, {
        children: "Extensions don't execute when locker is the swap sender"
      }), "\n", jsx(_components.li, {
        children: "Contact team if extension should be enabled but isn't"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mev-protection-issues",
      children: ["MEV Protection Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mev-protection-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": MEV protection not working as expected."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Diagnosis"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check MEV module status"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $HOOK "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"mevModuleEnabled(bytes32)(bool)"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " [POOL_ID] --rpc-url https://mainnet.base.org"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Check pool creation time"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "cast"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " call"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " $HOOK "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"poolCreationTimestamp(bytes32)(uint256)"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " [POOL_ID] --rpc-url https://mainnet.base.org"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# MEV protection expires after 2 minutes (120 seconds)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# If current time > creation + 120, protection is expired"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Expected Behavior"
      }), ":"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "MEV protection lasts maximum 2 minutes"
      }), "\n", jsx(_components.li, {
        children: "No-op module disables itself immediately"
      }), "\n", jsx(_components.li, {
        children: "Custom MEV modules can disable themselves early"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "liquidity-addition-blocked",
      children: ["Liquidity Addition Blocked", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#liquidity-addition-blocked",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Cannot add liquidity to pool."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Cause"
      }), ": MEV module is still active and blocking liquidity changes."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solution"
      }), ": Wait for MEV protection to expire (2 minutes max) or for module to disable itself."]
    }), "\n", jsxs(_components.h2, {
      id: "integration-issues",
      children: ["Integration Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "contract-interface-errors",
      children: ["Contract Interface Errors", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contract-interface-errors",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": ABI-related errors when calling contracts."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Use correct ABI"
        }), ": Ensure you're using the right ABI for each contract"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check function signatures"
        }), ": Verify function names and parameter types"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verify contract addresses"
        }), ": Use ", jsx(_components.a, {
          href: "/reference/addresses",
          children: "official addresses"
        })]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Example Correct Calls"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// ✅ Correct"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " deploymentInfo"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " await"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " factory."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "tokenDeploymentInfo"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(tokenAddress);"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// ❌ Wrong function name  "
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " deploymentInfo"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " await"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " factory."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "getTokenDeploymentInfo"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(tokenAddress);"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "gas-estimation-failures",
      children: ["Gas Estimation Failures", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#gas-estimation-failures",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Gas estimation failing for deployments."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Use higher gas limit"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "await"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " factory."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "deployToken"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(config, {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  gasLimit: "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "5_000_000"
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "  // High limit for deployments"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "});"
                })
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Check configuration validity"
          }), ": Invalid configs can cause gas estimation to fail"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Ensure sufficient ETH"
          }), ": Account must have enough ETH for gas + extension amounts"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "event-parsing-issues",
      children: ["Event Parsing Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#event-parsing-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Cannot parse contract events properly."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Common Issues"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Wrong event signature"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ✅ Correct"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "factory."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "on"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "'TokenCreated'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", ("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "sender"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "tokenAddress"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "..."
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "args"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "=>"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " {"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "  // Handle event"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "});"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// ❌ Wrong event name"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "factory."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "on"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "'TokenDeployed'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "..."
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "); "
                }), jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Event doesn't exist"
                })]
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Missing ABI"
          }), ": Ensure contract interface includes all events"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Block range too large"
          }), ": Limit historical queries to reasonable ranges"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "network-and-rpc-issues",
      children: ["Network and RPC Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#network-and-rpc-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "rpc-rate-limiting",
      children: ["RPC Rate Limiting", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#rpc-rate-limiting",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Getting rate limited by RPC provider."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Solutions"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Use multiple RPC endpoints"
          }), ":"]
        }), "\n", jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "const"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " fallbackProviders"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " ="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " ["
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "  'https://mainnet.base.org'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ","
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: "  'https://base-mainnet.public.blastapi.io'"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ","
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "  // Add more as backup"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "];"
                })
              })]
            })
          })
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Implement request caching"
          }), ": Cache responses to reduce RPC calls"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Add delays between requests"
          }), ": Space out rapid successive calls"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "transaction-failures",
      children: ["Transaction Failures", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#transaction-failures",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Problem"
      }), ": Transactions failing with unclear errors."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Debugging Steps"
      }), ":"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check transaction receipt"
        }), ": Look for specific revert reason"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Simulate transaction"
        }), ": Use ", jsx(_components.code, {
          children: "staticCall"
        }), " to debug before sending"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verify account state"
        }), ": Ensure sufficient balance and nonces are correct"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Check gas price"
        }), ": Ensure gas price is appropriate for network conditions"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "getting-help",
      children: ["Getting Help", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#getting-help",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "community-support",
      children: ["Community Support", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#community-support",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Discord"
      }), ": ", jsx(_components.a, {
        href: "https://discord.gg/feyprotocol",
        children: "Join our community"
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Real-time help from community"
      }), "\n", jsx(_components.li, {
        children: "Developer discussions"
      }), "\n", jsx(_components.li, {
        children: "Troubleshooting assistance"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "GitHub"
      }), ": ", jsx(_components.a, {
        href: "https://github.com/feyprotocol",
        children: "FEY Protocol Issues"
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Bug reports"
      }), "\n", jsx(_components.li, {
        children: "Feature requests"
      }), "\n", jsx(_components.li, {
        children: "Technical discussions"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Twitter"
      }), ": ", jsx(_components.a, {
        href: "https://twitter.com/feyprotocol",
        children: "@feyprotocol"
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Protocol announcements"
      }), "\n", jsx(_components.li, {
        children: "Status updates"
      }), "\n", jsx(_components.li, {
        children: "Community updates"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "reporting-bugs",
      children: ["Reporting Bugs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#reporting-bugs",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When reporting issues, please include:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Clear description"
        }), " of the problem"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Steps to reproduce"
        }), " the issue"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Expected vs actual behavior"
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Error messages"
        }), " (full stack trace if available)"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Environment details"
        }), " (browser, wallet, etc.)"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Transaction hashes"
        }), " for on-chain issues"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Template"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Problem**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": Brief description"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Steps to Reproduce**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ":"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "1."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Step 1"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "2."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Step 2"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "3."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Error occurs"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Expected**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": What should happen"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Actual**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": What actually happens"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Error Message**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            })]
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "Error details here"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Environment**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ":"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "-"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Browser: Chrome 120"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "-"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Wallet: MetaMask 11.0"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "-"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Transaction: 0x123..."
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                fontWeight: "bold",
                "--shiki-dark": "#ADBAC7",
                "--shiki-dark-font-weight": "bold"
              },
              children: "**Additional Context**"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": Any other relevant information"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "emergency-contacts",
      children: ["Emergency Contacts", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#emergency-contacts",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Security Issues"
      }), ": ", jsx(_components.a, {
        href: "mailto:security@feyprotocol.xyz",
        children: "security@feyprotocol.xyz"
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Critical vulnerabilities"
      }), "\n", jsx(_components.li, {
        children: "Potential exploits"
      }), "\n", jsx(_components.li, {
        children: "Responsible disclosure"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Critical Bugs"
      }), ': Use GitHub issues with "critical" label']
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Protocol-breaking issues"
      }), "\n", jsx(_components.li, {
        children: "Funds at risk scenarios"
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "prevention-tips",
      children: ["Prevention Tips", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#prevention-tips",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "before-deployment",
      children: ["Before Deployment", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#before-deployment",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      className: "contains-task-list",
      children: ["\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Validate all configuration arrays have correct lengths"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Ensure reward/position BPS sum to exactly 10000"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Verify extension addresses and data encoding"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Test on small amounts first"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Check factory and protocol status"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "during-integration",
      children: ["During Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#during-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      className: "contains-task-list",
      children: ["\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Use official contract addresses"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Implement proper error handling"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Add transaction confirmations"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Cache frequently accessed data"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Monitor for events and errors"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "monitoring",
      children: ["Monitoring", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#monitoring",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      className: "contains-task-list",
      children: ["\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Set up event listeners for critical functions"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Monitor fee accumulation and distribution"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Track protocol status changes"]
      }), "\n", jsxs(_components.li, {
        className: "task-list-item",
        children: [jsx(_components.input, {
          type: "checkbox",
          disabled: true
        }), " ", "Alert on failed transactions"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Related Documentation"
      }), ":"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/reference/addresses",
          children: "Contract Addresses →"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/guides/integration",
          children: "Integration Guide →"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/guides/cast-commands",
          children: "Cast Commands →"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "/support/security",
          children: "Security Guide →"
        })
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
